File: P5/QytetetRuby/pkg/QytetetRuby-0.0.1/lib/casilla.rb
#encoding:utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative 'jugador'
require_relative 'titulo_propiedad'

module ModeloQytetet
  class Casilla
    
    #Consultores
    attr_accessor :num_casilla, :coste, :tipo, :titulo
    
    def initialize (num, tipo, coste)
      @num_casilla = num
     # @titulo = tit
      @coste = coste
      @tipo = tipo
    end 
    
    
    
    def to_s
      str = ""
      if @tipo == TipoCasilla::CALLE
        str = " \n         Numero de casilla: #{@num_casilla} 
         Tipo: #{@tipo}
         #{@titulo.to_s()}  \n\n\n"
      end
      if @tipo != TipoCasilla::CALLE
        str = " \n        Numero de casilla: #{@num_casilla}
        Tipo: #{@tipo}"
      end
      
      return str
       
      
    end
  
    
    def soyEdificable()
        return false
    end
    
    
  end
end

File: P5/QytetetJava/src/modeloqytetet/Casilla.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

/**
 *
 * @author romanlarrosa
 */
public abstract class Casilla {
    private int numeroCasilla;
    private int coste;
//    private TipoCasilla tipo;
//    private TituloPropiedad titulo;
    
    public Casilla(int num, int _coste){
        numeroCasilla = num; 
        coste = _coste;
    }
    

    int getNumeroCasilla() {
        return numeroCasilla;
    }

    int getCoste() {
        return coste;
    }
    
    public void setCoste(int _coste){
        coste = _coste;
    }

    protected abstract TipoCasilla getTipo();

    protected abstract TituloPropiedad getTitulo();


    @Override
    public String toString() {
        return "\nCasilla numero " + numeroCasilla + 
                "\n Coste = " + coste; 
    }

    protected boolean soyEdificable(){
        return false;
    }
    
    
    
    
}
